apiserver/apiserver.go

```diff
@@ -607,6 +522,93 @@ func (srv *Server) mongoPinger() error {
 	}
 }
 
+// changeCertListener wraps a TLS net.Listener.
+// It allows connection handshakes to be
+// blocked while the TLS certificate is updated.
+type changeCertListener struct {
+	net.Listener
+
+	// srv holds the server that the listener was created by.
+	srv *Server
+
+	// certChanged is used to pass in new certificate information.
+	certChanged <-chan params.StateServingInfo
+
+	// mu guards the fields below it.
+	mu sync.Mutex
+
+	// cert holds the current certificate used for tls.Config
+	cert tls.Certificate
+}
+
+func (srv *Server) newChangeCertListener(lis net.Listener, certChanged <-chan params.StateServingInfo, cert tls.Certificate) *changeCertListener {
+	return &changeCertListener{
+		Listener:    lis,
+		srv:         srv,
+		cert:        cert,
+		certChanged: certChanged,
+	}
+}
+
+// Accept waits for and returns the next connection to the listener.
+func (cl *changeCertListener) Accept() (net.Conn, error) {
+	conn, err := cl.Listener.Accept()
+	if err != nil {
+		return nil, err
+	}
+	return tls.Server(conn, cl.tlsConfig()), nil
+}
+
+func (cl *changeCertListener) tlsConfig() *tls.Config {
+	cl.mu.Lock()
+	defer cl.mu.Unlock()
+	tlsConfig := utils.SecureTLSConfig()
+	//tlsConfig.GetCertificate = cl.getCertificate
```

Comment:
I always find commented out code disturbing, if it shouldn't be here could you remove it properly please?